tutorials/dataframe/df106_HiggsToFourLeptons_withVariations.py

```diff
@@ -118,53 +130,115 @@
     histos.append(df.Filter(f"sample_category == \"{sample_category}\"").Histo1D(ROOT.RDF.TH1DModel(f"{sample_category}", "m4l", 24, 80, 170), "m4l", "weight"))
 
 
-h_data = histos[0].GetValue()
-h_higgs = histos[1].GetValue()
-h_zz = histos[2].GetValue()
-h_other = histos[3].GetValue()
+# Evaluate the systematic uncertainty
 
-# Apply MC correction for ZZ due to missing gg->ZZ process
-h_zz.Scale(1.3) 
+# The systematic uncertainty in this analysis is the MC scale factor uncertainty that depends on lepton 
+# kinematics such as pT or pseudorapidity. 
+# Muons uncertainties are negligible, as stated in https://atlas.web.cern.ch/Atlas/GROUPS/PHYSICS/PAPERS/MUON-2018-03/. 
+# Electrons uncertainties are evaluated based on the plots available in https://doi.org/10.48550/arXiv.1908.00005. 
+# The uncertainties are linearly interpolated to cover a range of pT values covered by the analysis. 
 
-# Create the plot
+# Create a VaryHelper to interpolate the available data.
+ROOT.gInterpreter.Declare("""   
+#include <Math/Interpolator.h>                                   
+using namespace ROOT::VecOps;
+using namespace ROOT::Math;
+
+class VaryHelper 
+{
+    std::vector<double> x = {5.50 * 10e2, 5.52 * 10e2, 12.54 * 10e2, 17.43 * 10e2, 22.40 * 10e2, 27.48 * 10e2, 30 * 10e2, 10000 * 10e2};
+    std::vector<double> y = {0.06628, 0.06395, 0.06396, 0.03372, 0.02441, 0.01403, 0., 0.};
+    Interpolator inter;
+public:
+    VaryHelper() : inter(x, y, Interpolation::kLINEAR) {}
+
+    RVec<double> operator()(const double &w, RVec<float> &pt, RVec<unsigned int> &type)
+    {
+        const auto v = Mean(Map(pt[type == 11], [this](auto p) 
+        {return this->inter.Eval(p); })
+        );                                                                                              
+        return RVec<double>{(1 + v) * w, (1 - v) * w};
+    }
+};
+
+VaryHelper myHelper;
```

Comment:
```suggestion
VaryHelper myVaryHelper;
```
I think `myHelper` is a bit generic, here is an alternative suggestion. Choosing good variable names in the tutorials goes a long way when teaching good practices to users, and it's important in particular for more complicated analyses!